node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -737,6 +859,331 @@ mod tests {
+        let dir_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(home_dir.clone()))
+            .data_dir_result(Some(data_dir.to_path_buf()));
+
+        let result = server_initializer_collected_params(&dir_wrapper, args_vec.as_slice());
+        let env_multiconfig = result.unwrap();
+
+        #[cfg(not(target_os = "windows"))]
+        {
+            assert_eq!(
+                value_m!(env_multiconfig, "data-directory", String).unwrap(),
+                "booga/data_dir/MASQ/polygon-mainnet".to_string()
+            );
+            assert_eq!(
+                value_m!(env_multiconfig, "config-file", String).unwrap(),
+                current_dir().expect("expected curerrnt dir")
+                    .join(PathBuf::from( "./generated/test/node_configurator_standard/multi_config_vcl_is_computed_do_right_job/config.toml"))
+                    .to_string_lossy().to_string()
+            );
+        }
+        #[cfg(target_os = "windows")]
+        assert_eq!(
+            value_m!(env_multiconfig, "data-directory", String).unwrap(),
+            "generated/test/node_configurator_standard/multi_config_vcl_is_computed_do_right_job/home\\data_dir\\MASQ\\polygon-mainnet".to_string()
+        );
+    }
+
+    #[test]
+    fn server_initializer_collected_params_handle_dot_in_path_and_real_user_in_config_file() {
+        running_test();
+        let _guard = EnvironmentGuard::new();
+        let _clap_guard = ClapGuard::new();
+        let home_dir = ensure_node_home_directory_exists(
+            "node_configurator_standard",
+            "server_initializer_collected_params_handle_dot_in_path_and_real_user_in_config_file",
+        );
+        let data_dir = &home_dir.join("data_dir");
+        let config_file_relative = File::create(PathBuf::from("./generated/test/node_configurator_standard/server_initializer_collected_params_handle_dot_in_path_and_real_user_in_config_file").join("config.toml")).unwrap();
+        fill_up_config_file(config_file_relative);
+        let env_vec_array = vec![
+            ("MASQ_CONFIG_FILE", "./generated/test/node_configurator_standard/server_initializer_collected_params_handle_dot_in_path_and_real_user_in_config_file/config.toml"),
+            #[cfg(not(target_os = "windows"))]
+            ("MASQ_REAL_USER", "9999:9999:booga"),
+        ];
+        env_vec_array
+            .clone()
+            .into_iter()
+            .for_each(|(name, value)| std::env::set_var(name, value));
+        let args = ArgsBuilder::new();
+        let args_vec: Vec<String> = args.into();
+        let dir_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(home_dir.clone()))
+            .data_dir_result(Some(data_dir.to_path_buf()));
+
+        let result = server_initializer_collected_params(&dir_wrapper, args_vec.as_slice());
+        let env_multiconfig = result.unwrap();
+
+        assert_eq!(env_multiconfig.is_user_specified("--data-directory"), false);
```

Comment:
These assertions look to me to be very similar--maybe identical--to the assertions in the test above. So I wonder: how are these tests different? Are both required?

Reply:
this test is testing the scenario, when we specify the `config-file` with dot, assertions just proving that we correct the path correctly and we can read from config file - so `real-user` and `dns-servers` are asserted right way